### PR TITLE
Added documentation for TypeScript

### PR DESCRIPTION
**Pull request by megamindbrian**
The progress and dot reporters are great if you are just looking for a simple yes or no for breaking changes.  The spec reporter is useful because it lists the actual names of the test and whether it passes or fails, giving you a more classic experience to testing when coming from other frameworks.  NUnit, DotCover all have similar feature included in the package.  Progress and dot reporters have zero usefulness while debugging tests/code being tested.  This pull request fixes that problem by including the spec-reporter by default and minimizing the number of arbitrary dependencies.

### CODE REVIEW

**Comment by megamindbrian** [comment]
I can't tell why tests are failing, why would it be installing a lower version of npm?

**Comment by megamindbrian** [comment]
Review? @dignifiedquire . I don't know how to fix travis.

**Comment by Florian-R** [comment]
There's o lot of unrelated changes in there. Seems you have mixed your previous PR on spec reporter with a new one on Angular docs.

File: README.md [hunk 1]
```diff
@@ -10,19 +10,6 @@ _real_ browsers.
...
-
-> For questions and support please use the mailing list or Gitter.
-> The issue tracker is for bug reports and feature discussions only.
-
-* Obligatory [documentation]
-* Quick questions:
-[![Gitter Chat](https://img.shields.io/badge/GITTER-join%20chat-green.svg?style=flat-square)](https://gitter.im/karma-runner/karma)
-* Longer questions: [Mailing List]
-* Bug reports [Issue Tracker]
-* Everything less than 140 characters: [@JsKarma] on Twitter
```
Comment (Florian-R) [inline README.md]: What the point of all this README edit? The original conversational tone was better IMHO.

File: README.md [hunk 2]
```diff
@@ -86,27 +73,35 @@ See [installation](https://karma-runner.github.io/latest/intro/installation.html
...
-## This is so great. I want to help.
-
-Please, see
-[contributing](https://karma-runner.github.io/latest/dev/contributing.html).
-
-
 ## Why did you create this?
 
-Throughout the development of [AngularJS], we've been using [JSTD] for
+Throughout the development of [AngularJS] and [Angular], we've been using [JSTD] for
```
Comment (Florian-R) [inline README.md]: Nope, Angular wasn't developed with JSTD.

File: README.md [hunk 3]
```diff
@@ -86,27 +73,35 @@ See [installation](https://karma-runner.github.io/latest/intro/installation.html
 See [configuration](https://karma-runner.github.io/latest/intro/configuration.html).
 
 
-## This is so great. I want to help.
-
-Please, see
-[contributing](https://karma-runner.github.io/latest/dev/contributing.html).
```
Comment (Florian-R) [inline README.md]: Why did you cut this?

File: docs/plus/06-angular.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+angular-cli.json
+```json
+{
+    "test": {
+        "karma": {
+            "config": "./karma.conf.js"
+        }
+    }
+}
+```
```
Comment (Florian-R) [inline docs/plus/06-angular.md]: Test is built-in with angular-cli, are you sure this is needed?

File: docs/plus/06-angular.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    files: [
+        {pattern: 'src/**/*.ts', included: false, watched: false, served: false},
+        {pattern: './karma-test-shim.js', watched: false},
+    ],
+
+    preprocessors: {
+        './karma-test-shim.js': ['webpack', 'sourcemap'],
+    },
+    
+    webpack: webpackConfig,
```
Comment (Florian-R) [inline docs/plus/06-angular.md]: Not clear what is webpackConfig

File: README.md (hunk 1 above)
Comment (megamindbrian) [inline README.md]: I consolidated the contribution section, similar to https://github.com/jupyterlab/jupyterlab/README.md

File: README.md (hunk 3 above)
Comment (megamindbrian) [inline README.md]: Contributing link moved down below.

**Comment by johnjbarton** [comment]
Just cleaning up.
Please open a new Pull Request for this goal. Thanks!

**Pull request closed without merging**